pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -127,45 +129,268 @@ func (a *index) WorkloadsCollection(
 	return Workloads
 }
 
-func (a *index) workloadEntryWorkloadBuilder(
+func MergedGlobalWorkloadsCollection(
+	clusters krt.Collection[Cluster],
+	podsByCluster krt.Index[cluster.ID, krt.Collection[config.ObjectWithCluster[*v1.Pod]]],
+	nodesByCluster krt.Index[cluster.ID, krt.Collection[config.ObjectWithCluster[Node]]],
+	meshConfig krt.Singleton[MeshConfig],
+	localAuthorizationPolicies krt.Collection[model.WorkloadAuthorization],
+	localPeerAuths krt.Collection[*securityclient.PeerAuthentication],
+	waypointsByCluster krt.Index[cluster.ID, krt.Collection[config.ObjectWithCluster[Waypoint]]],
+	workloadServices krt.Collection[model.ServiceInfo],
```

Comment:
So this is going to take the merged ServiceInfo, and run it across all pods in any clusters.

Consider:

Local service has selector `app=bar`. Remote service has selector `app=foo`. Remote pod has `app=bar`.

This will incorrectly match the service to the pod.

Reply:
Ack, so essentially we need the unmerged version at this point. That's doable

Reply:
more broadly it would help to understand the actual desired inputs and output at a high level. like this type of thing, the waypoints question, etc. 

The only thing we really talked about in the doc was making a WDS workload per network/service